### PR TITLE
[FLINK-1712] Remove "flink-staging" module

### PR DESCRIPTION
**Pull request by rmetzger**
This PR is based on (and contains) https://github.com/apache/flink/pull/1482. I addressed all comments from #1482 and reworked the quickstart example (with new screenshots).

### CODE REVIEW

**Comment by StephanEwen** [comment]
Can you summarize what the changes from this pull request are? The set of changes is too large for github to display.

**Comment by rmetzger** [comment]
I moved the following modules out of "flink-staging":
flink-tez (to flink-contrib)
flink-ml (to flink-libraries)
flink-table (to flink-libraries)
flink-fs-tests (to root)
flink-scala-shell (to root)


I renamed "flink-staging" to "flink-batch-connectors", which contains the following modules
-> flink-avro
-> flink-jdbc
-> flink-hadoop-compatibility
-> flink-hbase
-> flink-hcatalog

**Comment by StephanEwen** [comment]
That seems like a good design.

+1 to merge from my side

File: flink-examples/flink-examples-batch/pom.xml [hunk 1]
```diff
@@ -280,208 +246,155 @@ under the License.
 					
 					<!-- EnumTriangles Basic -->
 					<execution>
-						<id>EnumTrianglesBasic</id>
+						<id>EnumerateGraphTriangles</id>
```
Comment (fhueske) [inline flink-examples/flink-examples-batch/pom.xml]: Rename to `EnumTriangles` to be in sync with class name.

File: flink-examples/flink-examples-batch/pom.xml [hunk 2]
```diff
@@ -280,208 +246,155 @@ under the License.
...
 						<phase>package</phase>
 						<goals>
 							<goal>jar</goal>
 						</goals>
 						<configuration>
-							<classifier>EnumTrianglesBasic</classifier>
+							<classifier>EnumerateGraphTriangles</classifier>
```
Comment (fhueske) [inline flink-examples/flink-examples-batch/pom.xml]: Rename to `EnumTriangles` to be in sync with class name.

File: flink-examples/flink-examples-batch/pom.xml [hunk 3]
```diff
@@ -280,208 +246,155 @@ under the License.
...
-							<includes>
-								<include>**/scala/wordcount/WordCount.class</include>
-								<include>**/scala/wordcount/WordCount$*.class</include>
-								<include>**/java/wordcount/util/WordCountData.class</include>
-							</includes>
+						<configuration> 
+							<target>
+								<copy file="${project.basedir}/target/flink-examples-batch-${project.version}-KMeans.jar" tofile="${project.basedir}/target/KMeans.jar" />
+								<copy file="${project.basedir}/target/flink-examples-batch-${project.version}-ConnectedComponents.jar" tofile="${project.basedir}/target/ConnectedComponents.jar" />
+								<copy file="${project.basedir}/target/flink-examples-batch-${project.version}-EnumerateGraphTriangles.jar" tofile="${project.basedir}/target/EnumerateGraphTriangles.jar" />
```
Comment (fhueske) [inline flink-examples/flink-examples-batch/pom.xml]: Rename to `EnumTriangles.jar` to be in sync with class name.

**Comment by fhueske** [comment]
Looks good to me, except for the name of the Triangle Enumeration example jar file. 
Would be nice to have this merged soon. 
I'd like to start working on putting the Table API on top of Apache Calcite.

**Comment by rmetzger** [comment]
Thank you for the reviews ... I'll merge the PR
